spec/controllers/forgot_password_controller_spec.rb

```diff
@@ -3,12 +3,11 @@
 RSpec.describe ForgotPasswordController do
   describe '#show' do
     context 'email in session' do
-      it 'renders the page and deletes the email from the session' do
+      it 'renders the page' do
         session[:email] = 'test@example.com'
 
         get :show
 
-        expect(session[:email]).to be_nil
```

Comment:
yes, but now that the logic was re-shuffled it will only happen if we render the view (`view.model.email` is called). 

Since rendering the views makes the test slower, I added a spec around session deletion in the unit test for the view model.